website/docs/d/vapp.html.markdown

```diff
@@ -0,0 +1,59 @@
+---
+layout: "vcd"
+page_title: "vCloudDirector: vcd_vapp"
+sidebar_current: "docs-vcd-datasource-vapp"
+description: |-
+  Provides a vCloud Director vApp data source. This can be used to reference vApps.
+---
+
+# vcd\_vapp
+
+Provides a vCloud Director vApp data source. This can be used to reference vApps.
+
+Supported in provider *v2.5+*
+
+## Example Usage
+
+
+```hcl
+data "vcd_vapp" "test-tf" {
+  name             = "test-tf"
+  org              = "tf"
+  vdc              = "vdc-tf"
+}
+
+output "name" {
+  value = data.vcd_vapp.test-tf.name
+}
+
+output "description" {
+  value = data.vcd_vapp.test-tf.description
+}
+
+output "href" {
+  value = data.vcd_vapp.test-tf.href
+}
+
+output "status_text" {
+  value = data.vcd_vapp.test-tf.status_text
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) A unique name for the vApp
+* `org` - (Optional) The name of organization to use, optional if defined at provider level. Useful when connected as sysadmin working across different organisations
+* `vdc` - (Optional) The name of VDC to use, optional if defined at provider level
+
+## Attribute reference
+
+* `href` - The vApp Hyper Reference
+* `metadata` -  Key value map of metadata to assign to this vApp. Key and value can be any string. (Since *v2.2+* metadata is added directly to vApp instead of first VM in vApp)
+* `power_on` -  A boolean value stating if this vApp should be powered on. Default is `true`
+* `guest_properties` -  Key value map of vApp guest properties **Note** `ovf` attribute
+sets guest properties on the first VM using a legacy ability of this resource to spawn 1 VM. Please
+use resources `vcd_vapp_vm` to provision VMs.
```

Comment:
It's good that we have this explanation, but after reading it it's still not clear what this `guest_properties` represent. Is it guest properties of the first VM? Is it the deprecated code that is setting it? If so, why are we returning it through vApp as opposed to through VM data source (we agreed not to burden data source with deprecated fields)?

Reply:
`guest properties` are assigned to the vApp, and any VM we add after that will inherit them.
This is not deprecated, and should not be

Reply:
`guest_properties` is the new one. `ovf` was the old one. This "propagation" to VMs is not done by us, this is how vCD behaves automatically.